src/workerd/jsg/jsg.h

```diff
@@ -703,7 +703,7 @@ class Data {
     destroy();
   }
   Data(Data&& other): isolate(other.isolate), handle(kj::mv(other.handle)) {
-    if (handle.IsWeak()) handle.ClearWeak();
+    if (other.tracedHandle != nullptr) handle.ClearWeak();
```

Comment:
Yeah, I suppose we could explicitly `Reset()` them and this may in fact allow minor GCs to be a bit more efficient, if they can see that all handles to an object have gone away.